integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,253 @@
+	file, err := os.ReadFile("./resources/config.json")
+	if err != nil {
+		return "", nil, err
+	}
+
+	err = json.Unmarshal(file, &cfgFileData)
+	if err != nil {
+		return "", nil, err
+	}
+
+	var logFiles []LogInfo
+	var logStreams []string
+
+	for i := 0; i < logNum; i++ {
+		logStream := fmt.Sprintf("{instance_id}/tmp%d", i + 1)
+
+		logFiles = append(logFiles, LogInfo {
+			FilePath: fmt.Sprintf("/tmp/test%d.log", i + 1),
+			LogGroupName: "{instance_id}",
+			LogStreamName: logStream,
+			Timezone: "UTC",
+		})
+
+		logStreams = append(logStreams, logStream)
+	}
+
+	log.Printf("Writing config file with %d logs to ./resources/config%d.json\n", logNum, logNum)
+
+	cfgFileData["logs"].(map[string]interface{})["logs_collected"].(map[string]interface{})["files"].(map[string]interface{})["collect_list"] = logFiles
+
+	finalConfig, err := json.MarshalIndent(cfgFileData, "", " ")
+	if err != nil {
+		return "", nil, err
+	}
+
+	configFilePath := fmt.Sprintf("./resources/config%d.json", logNum)
+	f, err := os.Create(configFilePath)
+	if err != nil {
+		return "", nil, err
+	}
+
+	defer f.Close()
+
+	f.Write(finalConfig)
+
+	return configFilePath, logStreams, nil
+}
+
+//StartLogWrite starts go routines to write logs to each of the logs that are monitored by CW Agent according to
+//the config provided
+func StartLogWrite(agentRunDuration time.Duration, configFilePath string, tps int) (error) {
+	//create wait group so main test thread waits for log writing to finish before stopping agent and collecting data
+	var logWaitGroup sync.WaitGroup
+
+	logPaths, err := GetLogFilePaths(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	for _, logPath := range logPaths {
+		filePath := logPath //necessary weird golang thing
+		logWaitGroup.Add(1)
+		go func() {
+			defer logWaitGroup.Done()
+			err = WriteToLogs(filePath, agentRunDuration, tps)
+		}()
+	}
+
+	//wait until writing to logs finishes
+	logWaitGroup.Wait()
+	return err
+}
+
+//WriteToLogs opens a file at the specified file path and writes the specified number of lines per second (tps)
+//for the specified duration
+func WriteToLogs(filePath string, durationMinutes time.Duration, tps int) (error) {
+	f, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	defer os.Remove(filePath)
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	endTimeout := time.After(durationMinutes)
+
+	//loop until the test duration is reached
+	for {
+		select {
+		case <-ticker.C:
+			for i := 0; i < tps; i++ {
+				_, err = f.WriteString(fmt.Sprintln(ticker, " - #", i, " This is a log line."))
+				if err != nil {
+					return err
+				}
+			}
+		
+		case <-endTimeout:
+			return nil
+		}
+	}
+}
+
+//GetLogFilePaths parses the cloudwatch agent config at the specified path and returns a list of the log files that the 
+//agent will monitor when using that config file
+func GetLogFilePaths(configPath string) ([]string, error) {
```

Comment:
Not worth blocking this PR for, but I think that this function can be pulled out of this test cod and into a utility.